zhaquirks/xiaomi/aqara/magnet_ac01.py

```diff
@@ -0,0 +1,75 @@
+"""Xiaomi aqara P1 contact sensor device."""
+
+from zigpy.profiles import zha
+import zigpy.types as t
+from zigpy.zcl.clusters.general import Basic, Identify, OnOff, Ota, PowerConfiguration
+from zigpy.zcl.clusters.security import IasZone
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.xiaomi import (
+    LUMI,
+    BasicCluster,
+    XiaomiAqaraE1Cluster,
+    XiaomiCustomDevice,
+)
+
+
+class OppleCluster(XiaomiAqaraE1Cluster):
+    """Opple cluster."""
+
+    class DetectionDistance(t.enum8):
+        """Detection distance."""
+
+        TenMM = 0x01
+        TwentyMM = 0x02
+        ThirtyMM = 0x03
+
+    attributes = {
+        0x010C: ("detection_distance", DetectionDistance, True),
+    }
+
+
+class LumiMagnetAC01(XiaomiCustomDevice):
+    """Custom device representing lumi.magnet.ac01."""
+
+    signature = {
+        # <SimpleDescriptor endpoint=1 profile=260 device_type=1026
+        # device_version=1
+        # input_clusters=[0, 1, 3, 1280]
+        # output_clusters=[25]>
+        MODELS_INFO: [(LUMI, "lumi.magnet.ac01")],
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.IAS_ZONE,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    PowerConfiguration.cluster_id,
+                    Identify.cluster_id,
+                    IasZone.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [Ota.cluster_id],
+            },
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                INPUT_CLUSTERS: [
+                    BasicCluster,
+                    PowerConfiguration.cluster_id,
```

Comment:
This might need to be replaced with `XiaomiPowerConfiguration` for proper battery measurements to show up(?)

Reply:
![image](https://github.com/zigpy/zha-device-handlers/assets/1335687/21cf549e-2727-44bd-830e-007f499df3af)

works OOTB it reports % and has correct battery information

Reply:
or maybe not :D I have to open it. It may actually be a 123a and not CR2